libarchive/archive_read_support_format_tar.c

```diff
@@ -763,7 +759,7 @@ tar_read_header(struct archive_read *a, struct tar *tar,
 					return (ARCHIVE_EOF);
 				}
 			}
-			if (bytes < 512) {  /* Short block at EOF; this is bad. */
+			if (h == NULL) {  /* Short block at EOF; this is bad. */
```

Comment:
This change has no effect but makes this part more consistent with the pattern now used throughout this source file.